Branch-SDK/Branch-SDK/BNCSystemObserver.m

```diff
@@ -131,23 +131,34 @@ + (NSNumber *)getUpdateState {
     // for creation date
     NSURL *documentsDirRoot = [[[NSFileManager defaultManager] URLsForDirectory:NSDocumentDirectory inDomains:NSUserDomainMask] lastObject];
     NSDictionary *documentsDirAttributes = [manager attributesOfItemAtPath:documentsDirRoot.path error:nil];
-    int appCreationDay = (int)([[documentsDirAttributes fileCreationDate] timeIntervalSince1970]/(60*60*24));
-
+    NSDate *creationDate = [documentsDirAttributes fileCreationDate];
+    
     // for modification date
     NSString *bundleRoot = [[NSBundle mainBundle] bundlePath];
     NSDictionary *bundleAttributes = [manager attributesOfItemAtPath:bundleRoot error:nil];
-    int appModificationDay = (int)([[bundleAttributes fileModificationDate] timeIntervalSince1970]/(60*60*24));
-
+    NSDate *modificationDate = [bundleAttributes fileModificationDate];
+    
+    // No stored version
     if (!storedAppVersion) {
-        if ([documentsDirAttributes fileCreationDate] && [bundleAttributes fileModificationDate] && (appCreationDay != appModificationDay)) {
-            return [NSNumber numberWithInt:2];
+        // Modification and Creation date are more than 60 seconds different indicates an update
+        // This would be the case that they had un-installed and re-installed the app, since this
+        // value isn't currently in NSDefaults.
+        if (creationDate && modificationDate && [modificationDate timeIntervalSinceDate:creationDate] > 60) {
```

Comment:
I also considered that, but came to the same conclusion -- the modification date of the bundle couldn't be before the creation date of the the documents directory, right?